pkg/update/root.go

```diff
@@ -75,6 +79,13 @@ func (c *RootCommand) Exec(in io.Reader, out io.Writer) error {
 		}
 	}
 
+	c.Globals.File.CLI.Version = latest.String()
+
+	// Write the file data to disk.
+	if err := c.Globals.File.Write(c.configFilePath); err != nil {
+		return fmt.Errorf("error saving config file: %w", err)
+	}
+
```

Comment:
IIUC by the time we reach this code we've already fetched the latest config. 
Is this code to avoid refetching it after the update?

Reply:
Correct! The scenario here is this:

- User runs `fastly update` and the config is (potentially) fetched from the remote endpoint. 
  - The local config is now cached for 5 minutes.
  - We write the updated version back to the local configuration file.
- User runs a new command, config won't be re-fetched because it's TTL hasn't expired.
  - But now in `main.go` we have a check for `file.CLI.Version` to see if it matches what's baked into the binary (i.e. `revision.AppVersion`), so we see the config has the same version as the running binary and so we avoid re-fetching the remote configuration.

Reply:
Thank you.

Personally I'd prefer to remove this code and fetch it again for better encapsulation but ymmv.